stream_read.py

```diff
@@ -0,0 +1,383 @@
+                        # I found a line `bstar`
+                        flags["cstar"] = True
+                        # creat list
+                        cstar = [float(x) for x in line.split(' ')[2:-1]]
+                    elif "lattice_type" in line:
+                        # I found a line `lattice_type`
+                        flags["lattice_type"] = True
+                        lattice_type = line.strip().split(' ')[2]
+                    elif "centering" in line:
+                        # I found a line `centering`
+                        flags["centering"] = True
+                        centering = line.strip().split(' ')[2]
+                    elif "unique_axis" in line:
+                        # I found a line `unique_axis`
+                        flags["unique_axis"] = True
+                        unique_axis = line.strip().split(' ')[2]
+
+                if "--- End crystal" in line:
+                        # the end line of cryst. info.
+                    # We need `astar`, `bstar` and `cstar`
+                    # to calculate unit cell parameters
+                    if not(flags["astar"] or flags["bstar"] or flags["cstar"]):
+                        LOGGER.warning("Image {} has bad cell".format(name))
+                    else:
+                        a, b, c, alfa, beta, gamma =\
+                            cell_parameters(astar, bstar, cstar)
+                        if not (flags["lattice_type"] and
+                                flags["centering"] and flags["unique_axis"]):
+
+                            # if I do not have `lattice_type` ,`centering`
+                            # or `unique_axis`then
+                            # I keep the default (`triclinic` `P` , `?`)
+                            LOGGER.warning(
+                                "{} keep default triclinic P".format(name))
+                            lattice_type = "triclinic"
+                            centering = "P"
+                            unique_axis = "?"
+
+                        crystal = {'name': name, 'a': a, 'b': b, 'c': c,
+                                   'alfa': alfa, 'beta': beta,
+                                   'gamma': gamma, 'centering': centering,
+                                   'lattice_type': lattice_type,
+                                   'unique_axis': unique_axis}
+                        crystals.append(crystal)
+                        # reset event_name
+                        event_name = ""
+                        # reset flags
+                        for key in flags.keys():
+                            flags[key] = False
+    except TypeError:
+        LOGGER.critical("Wrong path to the stream file.")
+        sys.exit()
+    except IndexError:
+        LOGGER.critical("Enter path to the stream file.")
+        sys.exit()
+    except FileNotFoundError:
+        LOGGER.critical("File not found or not a indexing stream file.")
+        sys.exit()
```

Comment:
Same as a previous comment: use sys.exit(1) to indicate that the exit is due to a problem.

Alternatively, rethrow the same error with raise

Reply:
Addressed in: b2c5306